lib/internal/process/next_tick.js

```diff
@@ -32,32 +32,53 @@ function setupNextTick() {
   const kHasScheduled = 0;
   const kHasPromiseRejections = 1;
 
-  const nextTickQueue = {
-    head: null,
-    tail: null,
+  // Queue size for each tick array. Must be a factor of two.
+  const kQueueSize = 2048;
+  const kQueueMask = kQueueSize - 1;
+
+  class FixedQueue {
+    constructor() {
+      this.btm = 0;
+      this.top = 0;
+      this.list = new Array(kQueueSize);
+      this.next = null;
+    }
+
     push(data) {
-      const entry = { data, next: null };
-      if (this.tail !== null) {
-        this.tail.next = entry;
-      } else {
-        this.head = entry;
-        tickInfo[kHasScheduled] = 1;
-      }
-      this.tail = entry;
-    },
+      this.list[this.top] = data;
+      this.top = (this.top + 1) & kQueueMask;
+    }
+
     shift() {
-      if (this.head === null)
-        return;
-      const ret = this.head.data;
-      if (this.head === this.tail) {
-        this.head = this.tail = null;
+      var next = this.list[this.btm];
```

Comment:
You can safely use `const` here.

Reply:
Thanks, was wondering if that came with a perf cost.